openlibrary/templates/work_search.html

```diff
@@ -202,10 +202,18 @@ <h1>$_("Search Books")</h1>
                 $ works = add_read_statuses(username, works)
             $ show_librarian_extras = ctx.user and (ctx.user.is_admin() or ctx.user.is_usergroup_member('/usergroup/librarians'))
 
-            $for work in works:
-                $ read_status = work.get('readinglog', None)
-                $ dropper = macros.ReadingLogDropper([], work=work, include_lists=False, page_url="/search", users_work_read_status=read_status)
-                $:macros.SearchResultsWork(work, reading_log=dropper, show_librarian_extras=show_librarian_extras)
+            $if safe_mode:
+                $ subject_keys = [d['subject_key'] for d in docs]
+                $for work,subject_key in zip(works,subject_keys):
+                    $ read_status = work.get('readinglog', None)
+                    $ dropper = macros.ReadingLogDropper([], work=work, include_lists=False, page_url="/search", users_work_read_status=read_status)
+                    $ content_warning = len([s for s in subject_key if s.startswith("content_warning")]) != 0
+                    $:macros.SearchResultsWork(work, reading_log=dropper, show_librarian_extras=show_librarian_extras, blur=content_warning)
+            $else:
+                $for work in works:
+                    $ read_status = work.get('readinglog', None)
+                    $ dropper = macros.ReadingLogDropper([], work=work, include_lists=False, page_url="/search", users_work_read_status=read_status)
+                    $:macros.SearchResultsWork(work, reading_log=dropper, show_librarian_extras=show_librarian_extras, blur=False)
```

Comment:
```suggestion
            $ subject_keys = [d.get('subject_key', []) for d in docs)
            $for work, subjects in zip(works, subject_keys):
                $ read_status = work.get('readinglog', None)
                $ dropper = macros.ReadingLogDropper([], work=work, include_lists=False, page_url="/search", users_work_read_status=read_status)
                $ content_warning = len([s for s in subjects if s.startswith("content_warning")]) != 0
                $:macros.SearchResultsWork(work, reading_log=dropper, show_librarian_extras=show_librarian_extras, blur=content_warning)
```
A few changes here:
1. `subject_key` may not always be on the `doc`.  Use `get()` to default to an empty list for these cases.
2. Checking for `safe_mode` is not needed today, as the `subject_key` list will only be included in Solr results if the patron enabled safe mode.  If this changes in the future, `safe_mode` can be used to set `content_warning`'s value.
3. Some of this code is repeated, which makes maintenance a bit more challenging.